source/framework/core/inc/TRestHits.h

```diff
@@ -173,10 +173,11 @@ class TRestHits : public TObject {
     Double_t GetMinimumHitEnergy() const;
     Double_t GetMeanHitEnergy() const;
 
-    Double_t GetEnergyIntegral() const;
+    Double_t CalculateTotalDepositedEnergy();
```

Comment:
Personally I think this method name is misleading, I think all methods that return a result and do not alter anything (are marked `const`) should start by `Get`, in this case maybe it should be `GetTotalDepositedEnergy`. This method should be `const`.

In my opinion a method named like this would be expected to update some internal member with some expensive to calculate magnitude which could be later accessed via a getter, and this calculation is not directly implemented in the getter because its expensive.

Reply:
I agree with you, can be renamed but should be stated that is a `Getter` and should be `const`

Reply:
In fact is changing a data member `fTotEnergy`, so you can do
```
void CalculateTotalDepositedEnergy(){fTotEnergy = GetEnergyIntegral(); }
```

Reply:
> In fact is changing a data member fTotEnergy, so you can do

That's what I want. We lack a method that updates `fTotEnergy` data member, in case some processes modifies hit point directly.  I can make the method void in this way.